addons/product_configurator/static/src/js/product_configurator_dialog/product_configurator_dialog.js

```diff
@@ -19,16 +19,19 @@ export class ProductConfiguratorDialog extends Component {
                 value: String,
             }
         },
-        quantity: Number,
+        quantity: { type: Number, optional: true },
```

Comment:
do not make optional, give default 0.0 value and hide if hide_qty in options (extension in mrp_mps)